src/disco/gui/fd_gui_printf.c

```diff
@@ -0,0 +1,1130 @@
+  /* escape quotemark, reverse solidus, and control chars U+0000 through U+001F
+     just replace with a space */
+  uchar * data = http->oring;
+  for( ulong i=start_len; i<http->stage_len; i++ ) {
+    if( FD_UNLIKELY( data[ (http->stage_off%http->oring_sz)+i ] < 0x20 ||
+                     data[ (http->stage_off%http->oring_sz)+i ] == '"' ||
+                     data[ (http->stage_off%http->oring_sz)+i ] == '\\' ) ) {
+      data[ (http->stage_off%http->oring_sz)+i ] = ' ';
+    }
+  }
+}
+
+static void
+jsonp_string( fd_gui_t *   gui,
+              char const * key,
+              char const * value ) {
+  char * val = (void *)value;
+  if( FD_LIKELY( value ) ) {
+    if( FD_UNLIKELY( !fd_utf8_verify( value, strlen( value ) ) )) {
+      val = NULL;
+    }
+  }
+  if( FD_LIKELY( key ) ) fd_http_server_printf( gui->http, "\"%s\":", key );
+  if( FD_LIKELY( val ) ) {
+    fd_http_server_printf( gui->http, "\"" );
+    ulong start_len = gui->http->stage_len;
+    fd_http_server_printf( gui->http, "%s", val );
+    jsonp_sanitize_str( gui->http, start_len );
+    fd_http_server_printf( gui->http, "\"," );
+  } else {
+    fd_http_server_printf( gui->http, "null," );
+  }
+}
+
+static void
+jsonp_bool( fd_gui_t *   gui,
+            char const * key,
+            int          value ) {
+  if( FD_LIKELY( key ) ) fd_http_server_printf( gui->http, "\"%s\":%s,", key, value ? "true" : "false" );
+  else                   fd_http_server_printf( gui->http, "%s,", value ? "true" : "false" );
+}
+
+static void
+jsonp_null( fd_gui_t *   gui,
+            char const * key ) {
+  if( FD_LIKELY( key ) ) fd_http_server_printf( gui->http, "\"%s\": null,", key );
+  else                   fd_http_server_printf( gui->http, "null," );
+}
+
+static void
+jsonp_open_envelope( fd_gui_t *   gui,
+                     char const * topic,
+                     char const * key ) {
+  jsonp_open_object( gui, NULL );
+  jsonp_string( gui, "topic", topic );
+  jsonp_string( gui, "key",   key );
+}
+
+static void
+jsonp_close_envelope( fd_gui_t * gui ) {
+  jsonp_close_object( gui );
+  jsonp_strip_trailing_comma( gui );
+}
+
+void
+fd_gui_printf_open_query_response_envelope( fd_gui_t *   gui,
+                                            char const * topic,
+                                            char const * key,
+                                            ulong        id ) {
+  jsonp_open_object( gui, NULL );
+  jsonp_string( gui, "topic", topic );
+  jsonp_string( gui, "key", key );
+  jsonp_ulong( gui, "id", id );
+}
+
+void
+fd_gui_printf_close_query_response_envelope( fd_gui_t * gui ) {
+  jsonp_close_object( gui );
+  jsonp_strip_trailing_comma( gui );
+}
+
+void
+fd_gui_printf_null_query_response( fd_gui_t *   gui,
+                                   char const * topic,
+                                   char const * key,
+                                   ulong        id ) {
+  jsonp_open_object( gui, NULL );
```

Comment:
You can use `fd_gui_printf_open_query_response_envelope` and `fd_gui_printf_close_query_response_envelope` here